UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Validations.cs

```diff
@@ -750,6 +750,19 @@ public static bool HasUsedActiveWelder(Interaction interaction)
 		return welder.IsOn;
 	}
 
+	/// <summary>
+	/// Checks if a welder is on and is being used, also works with blowtorches
+	/// </summary>
+	/// <param name="objectToCheck"></param>
+	/// <returns></returns>
+	public static bool HasUsedActiveWelder(GameObject objectToCheck)
+	{
+		if (objectToCheck == null) return false;
+		var welder = objectToCheck.GetComponent<FireSource>();
+		if (welder == null) return false;
+		return welder.IsBurning && HasItemTrait(objectToCheck, CommonTraits.Instance.Welder);
+	}
```

Comment:
Doing it this way means that blow torches can't be used for anything welders do currently.

I think the preferred solution would be to make the Welder script a base class in the main repo and then inherit two scripts which do the sprites the different ways?

Reply:
Yeah no this is a far better implementation, ive made the change